include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1167 @@
+                    $this->pullEvent($gevent, $meeting);
+                    break;
+                case "skip":
+                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Skipping Event: ' . $gevent->getSummary());
+                    break;
+                case "push_delete":
+                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Deleting Event: ' . $meeting->name);
+                    $this->delEvent($gevent, $meeting->id);
+                    break;
+                case "pull_delete":
+                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Deleting Meeting: ' . $meeting->name);
+                    $this->delMeeting($meeting);
+                    break;
+                default:
+                    $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'pushPullSkip() returned unknown value: ' . $dowhat . ' for event: ' . $gevent->getSummary());
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Setup array of users to sync
+     *
+     * Fills the $users array with users that are configured to sync
+     *
+     * @return null
+     */
+    public function setSyncUsers()
+    {
+
+        $query = "SELECT id FROM users WHERE deleted = '0'";
+        $result = $this->db->query($query);
+
+        require_once('modules/Users/User.php');
+        while ($row = $this->db->fetchByAssoc($result)) {
+            $user = new User();
+            $user->retrieve($row['id']);
+
+            if (!empty($user->getPreference('GoogleApiToken', 'GoogleSync')) &&
+            $accessToken = json_decode(base64_decode($user->getPreference('GoogleApiToken', 'GoogleSync'))) &&
+            $user->getPreference('syncGCal', 'GoogleSync') == '1') {
+                $this->addUser($user->id, $user->full_name);
+            }
+        }
+    }
+
+    /**
+     * Sync All Configured Users
+     *
+     * Running this method will collect all users who
+     * have Calendar Sync Configured and Enabled and
+     * Sync them one by one.
+     *
+     * @return bool Success/Failure
+     */
+    public function syncAllUsers()
+    {
+
+        // First we populate the array of syncable users
+        $this->setSyncUsers();
+
+        // We count failures here
+        $failures = 0;
+
+        // Then we go though the array and sync the users with doSync()
+        if (isset($this->users) && !empty($this->users)) {
+            foreach ($this->users as $key => $value) {
+                $return = $this->doSync($key);
+                if (!$return) {
+                    $GLOBALS['log']->error('Something went wrong syncing for user id: ' . $key);
+                    $failures++;
+                }
+            }
+        }
+        if ($failures > 0) {
+            return false;
```

Comment:
passing the $failures itself make more informative return value of this function.

Reply:
IIRC, this needs to return false on failure so that the cron job can be marked as failed.
I could have just done $failures = 1; instead of $failures++;